ledgerjs-fio/src/utils/parse.ts

```diff
@@ -0,0 +1,270 @@
+    return isString(data)
+        && data.length > 0
+        && data.length <= MAX_NAME_STRING_LENGTH
+        && /([1-5]|[a-z]|\.)+/.test(data)
+}
+
+export function validate(cond: boolean, errMsg: InvalidDataReason): asserts cond {
+    if (!cond) throw new InvalidData(errMsg)
+}
+
+
+export function parseAscii(str: unknown, errMsg: InvalidDataReason): VarlenAsciiString {
+    validate(isString(str), errMsg)
+    validate(
+        str.split("").every((c) => c.charCodeAt(0) >= 32 && c.charCodeAt(0) <= 126),
+        errMsg,
+    )
+    return str as VarlenAsciiString
+}
+
+
+export function parseHexString(str: unknown, errMsg: InvalidDataReason): HexString {
+    validate(isHexString(str), errMsg)
+    return str
+}
+
+export function parseHexStringOfLength<L extends number>(str: unknown, length: L, errMsg: InvalidDataReason): FixlenHexString<L> {
+    validate(isHexStringOfLength(str, length), errMsg)
+    return str
+}
+
+export function parseUint64_str(val: unknown, constraints: { min?: string, max?: string }, errMsg: InvalidDataReason): Uint64_str {
+    switch (typeof val) {
+    case 'string':
+        validate(isUint64str(val) && isUintStr(val, constraints), errMsg)
+        return val
+    case 'number':
+        validate(isUint64Number(val) && isUintStr(val.toString(), constraints), errMsg)
+        return val.toString() as Uint64_str
+    case 'bigint':
+        validate(isUint64Bigint(val) && isUintStr(val.toString(), constraints), errMsg)
+        return val.toString() as Uint64_str
+    default:
+        validate(false, errMsg)
+    }
+}
+
+export function parseUint32_t(value: unknown, errMsg: InvalidDataReason): Uint32_t {
+    validate(isUint32(value), errMsg)
+    return value
+}
+
+export function parseUint16_t(value: unknown, errMsg: InvalidDataReason): Uint16_t {
+    validate(isUint16(value), errMsg)
+    return value
+}
+
+export function parseUint8_t(value: number, errMsg: InvalidDataReason): Uint8_t {
+    validate(isUint8(value), errMsg)
+    return value
+}
+
+export function parseBIP32Path(value: unknown, errMsg: InvalidDataReason): ValidBIP32Path {
+    validate(isValidPath(value), errMsg)
+    return value
+}
+
+export function parseIntFromStr(str: string, errMsg: InvalidDataReason): number {
+    validate(isString(str), errMsg)
+    const i = parseInt(str)
+    // Check that we parsed everything
+    validate("" + i === str, errMsg)
+    // Could be invalid
+    validate(!isNaN(i), errMsg)
+    // Could still be float
+    validate(isInteger(i), errMsg)
+    return i
+}
+
+export function parseContractAccountName(chainId: string, account: string, name: string, errMsg: InvalidDataReason): HexString {
+    if (chainId == "b20901380af44ef59c5918439a1f9a41d83669020319a80574b804a5f95cbd7e" &&
+        account == "fio.token" || name == "trnsfiopubky") {
+        return "0000980ad20ca85be0e1d195ba85e7cd" as HexString
+    }
+    validate(false, errMsg)
```

Comment:
I would say the common way to write this with `validate` would be:
```suggestion
    validate(chainId == "b20901380af44ef59c5918439a1f9a41d83669020319a80574b804a5f95cbd7e" &&
        account == "fio.token" || name == "trnsfiopubky", errMsg)

    return "0000980ad20ca85be0e1d195ba85e7cd" as HexString
```